### PR TITLE
ROS2 Package for build nodes

### PR DESCRIPTION
**Pull request by amzn-changml**
## What does this PR do?

Adds the ROS2 package to the Ubuntu build nodes in Jenkins

In addition, cleanup of the install scripts and updates that were made out of band.

## How was this PR tested?

Tested in a sandbox Jenkins instance

### CODE REVIEW

**Review by brianherrera — APPROVED** [review]

**Review by byrcolin — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

File: scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh [hunk 1]
```diff
@@ -23,21 +23,25 @@ then
...
-elif [ "$UBUNTU_DISTRO" == "focal" ]
-then
-    echo "Setup for Ubuntu 20.04 LTS ($UBUNTU_DISTRO)"
-elif [ "$UBUNTU_DISTRO" == "jammy" ]
+#
+# Get Ubuntu version data
+#
+UBUNTU_DISTRO="$(lsb_release -sc)"
+UBUNTU_VER="$(lsb_release -sr)"
+if [ "$UBUNTU_DISTRO" == "bionic" ] || [ "$UBUNTU_DISTRO" == "focal" ] || [ "$UBUNTU_DISTRO" == "jammy" ]
```
Comment (michalpelka) [inline scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh:31]: Note that ROS2-Gem supports only Humble and Galactic ROS2. ROS2 Galactic is EOL, so only Ubuntu 22.04 and Humble should be supported for ROS2-Gem.

File: scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt [hunk 2]
```diff
@@ -21,3 +20,4 @@ libxcb-xfixes0-dev                      # For mouse input
 libxcb-xinput-dev                       # For mouse input
 zlib1g-dev
 mesa-common-dev
+ros-dev-tools                           # ROS2 development tools
```
Comment (michalpelka) [inline scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt:23]:
```suggestion
ros-humble-ackermann-msgs 
ros-humble-control-toolbox 
ros-humble-gazebo-msgs 
ros-humble-tf2-ros
ros-humble-urdfdom
ros-humble-vision-msgs
````

File: scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh (hunk 1 above)
Comment (amzn-changml) [reply scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh:31]: Good call out, I'll remove the package from the 20.04 (focal) package list

File: scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt (hunk 2 above)
Comment (amzn-changml) [reply scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt:23]: That's fair, are these the minimum sets or would a developer need more from `ros-dev-tools` to debug?

File: scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh (hunk 1 above)
Comment (amzn-changml) [reply scripts/build/build_node/Platform/Linux/install-ubuntu-build-tools.sh:31]: Actually on second thought, I'll leave it in for our CI processes, but we'll call out that it's not supported in docs

**Pull request merged**